### PR TITLE
Asciidoctor: Work around strange attribute rules

### PR DESCRIPTION
**Pull request by nik9000**
Asciidoc requires that we fence attributes at the beginning of a section
like so:
```
--
:api: index
:request: IndexRequest
:response: IndexResponse
--
```

But asciidoctor drops the attribute values if they are in the block
which I've filed here:
https://github.com/asciidoctor/asciidoctor/issues/2993

That means that we have to *remove* the blocks for asciidoctor. This
does that.

### CODE REVIEW

File: resources/asciidoctor/spec/spec_helper.rb [hunk 1]
```diff
@@ -28,16 +28,20 @@ def initialize warnings, result
...
   result = Asciidoctor.convert input, {
-      :safe => :unsafe,  # Used to include "funny" files.
-      :backend => :docbook45,
-      :logger => logger,
-      :doctype => :book,
+      :safe       => :unsafe,  # Used to include "funny" files.
+      :backend    => :docbook45,
+      :logger     => logger,
+      :doctype    => :book,
+      :verbose    => true,
```
Comment (ddillinger) [inline resources/asciidoctor/spec/spec_helper.rb]: Should this use the `$VERBOSE` variable?

Comment (nik9000) [reply resources/asciidoctor/spec/spec_helper.rb]: Oh! Actually it doesn't do anything. Let me drop it.

Comment (nik9000) [reply resources/asciidoctor/spec/spec_helper.rb]: I tried this first because it feels like it should work, but it doesn't. Because :shrug:

**Comment by nik9000** [comment]
> Do we have, or need to support, mixed blocks? I see tests for attribute-only, and non-attribute-having, but nothing for has-an-attribute-and-also-something-else (but I don't know if that's a thing that can happen...)

You can totally have mixed blocks and we should treat them like the non-attribute-having blocks because we want to run normal processing on them. Honestly I'd prefer not to do anything with blocks but this seems to be what asciidoctor needs right now but it is weird and I don't like it so I want to keep it to a minimum. I'll add a mixed test just in case though.

**Comment by nik9000** [comment]
>  I'll add a mixed test just in case though.

Done!

**Comment by nik9000** [comment]
Thanks for reviewing @ddillinger!

**Pull request merged**